### ISSUE TITLE
[i18n] option to check content of tag with i18n

### ISSUE THREAD

**Issue by superole** [angular-eslint/angular-eslint#930]
**Description and reproduction of the issue**
I would like to propose that you include an option to the i18n rule.

This option should make the rule check the content of a tag that has the i18n attribute set, and report if the content is not plain text (eg. it contains markup).

I recently discovered that a misplaced i18n attribute caused a bug in an application, because the markup content in the tag was replaced with plain text.

I try to enforce the convention that markup (and application logic) should not be mixed with i18n, and I would like this rule to help me with that. Therefore it is also my opinion that this new option should be enabled by default.

```JSON
{
  "rules": {
    "@angular-eslint/template/i18n": ["error"]
  }
}
```
OR
```JSON
{
  "rules": {
    "@angular-eslint/template/i18n": ["error",{"newOptionName": "true"}]
  }
}
```

**Invalid code for this rule**
```HTML
<div i18n="@@mainHeader">
  <h1>Hello i18n!</h1>
</div>
```
OR
```HTML
<!-- this was changed from a text-button to an icon with a tooltip, but we forgot to remove the i18n for the button text -->
<div i18n="@@buttonText" i18n-title="@@buttonTooltip" title="Enable magic">
  <fa-icon [icon]="['fas', 'magic']"></fa-icon>
</div>
```

**Valid code for this rule**
```HTML
<div>
  <h1 i18n="@@mainHeader">Hello i18n!</h1>
</div>
```
OR
```HTML
<div i18n-title="@@buttonTooltip" title="Enable magic">
  <fa-icon [icon]="['fas', 'magic']"></fa-icon>
</div>
```

**Comment by rajsite** [comment]
We ran into a very similar situation using web components with Angular and slotted content:

invalid:
```html
<my-button i18n>
    <my-icon slot="icon"></my-icon>
    Hello World
</my-button>
```

valid:
```html
<my-button>
    <my-icon slot="icon"></my-icon>
    <span i18n>Hello World</span>
</my-button>
```

The invalid case resulted in the icon html being considered part of the localized string which isn't what we really wanted and I expect few do.
